polkadb/dbService.go

```diff
@@ -68,14 +68,19 @@ func (s *DbService) Start() error {
 // Stop kills running BlockDB and StateDB instances
 func (s *DbService) Stop() error {
 	// Closing Badger Databases
-	err := s.StateDB.Db.Close()
-	if err != nil {
-		return err
+	if s.StateDB != nil {
```

Comment:
These shouldn't be necessary. Any successful call to Start will lead to successful call to Stop. If you need this you probably aren't using the service correctly. Would you mind elaborating on this?

Reply:
I noticed that one test was sometimes panicking without this, so I figured it would be nice to not have that. I could remove it though

Reply:
I tried removing these and running the test suite and nothing panicked. Please remove or let me know if you can reproduce.

Reply:
okay I removed it and it was fine, they are gone now